swaggerpy/swagger_model.py

```diff
@@ -307,35 +299,35 @@ def load_json(resource_listing, http_client=None, processors=None):
     return resource_listing
 
 
+class docstring_property(object):
+    def __init__(self, func):
+        self.func = func
+
+    def __get__(self, _cls, _owner):
+        return self.func()
+
+
 def create_model_type(model):
     """creates a dynamic model from the model data present in the json
-       :param model: Resource Model json containing id, properties
-       :type model: dict
-       :returns: dynamic type created with attributes, docstrings attached
-       :rtype: type
+
+    :param model: Resource model :class:`dict` with keys `id` and `properties`
+    :returns: dynamic type created with attributes, docstrings attached
+    :rtype: type
     """
     props = model['properties']
     name = str(model['id'])
+
     methods = dict(
-        # Magic Methods :
-        # Define the docstring
-        __doc__=create_model_docstring(props),
-        # Make equality work for dict & type OR type & type
+        __doc__=docstring_property(partial(create_model_docstring, props)),
         __eq__=lambda self, other: compare(self, other),
-        # Define the constructor for the type
         __init__=lambda self, **kwargs: set_props(self, **kwargs),
-        # Define the str repr of the type
         __repr__=lambda self: create_model_repr(self),
-        # Instance methods :
-        # Generates flat dict from the model instance
-        _flat_dict=lambda self: create_flat_dict(self))
-    model_type = type(name, (object,), methods)
-    # Define a class variable to store types of its attributes
-    setattr(model_type, '_swagger_types',
-            swagger_type.get_swagger_types(props))
-    # Define a class variable to store all required fields
-    setattr(model_type, '_required', model.get('required'))
-    return model_type
+        __dir__=lambda self: props.keys(),
+        _flat_dict=lambda self: create_flat_dict(self),
+        _swagger_types=swagger_type.get_swagger_types(props),
+        _required=model.get('required'),
+    )
+    return type(name, (object,), methods)
```

Comment:
I think you need to call out in simple terms why we're doing this in the first place and the net effect: performance reasons and lazy docstring generation in REPL